src/libs/actions/Task.ts

```diff
@@ -1266,6 +1274,19 @@ function clearTaskErrors(reportID: string | undefined) {
     });
 }
 
+function getFinishOnboardingTaskOnyxData(taskName: keyof OnyxTypes.IntroSelected): OnyxData {
```

Comment:
@thelullabyy Would it be possible to make this into a pure function? The way it is now, it is accessing data from two globals:

- introSelected
- allReports

In my PR https://github.com/Expensify/App/pull/60869 I am having to add one more due to the way this function was initially written:

- reportNameValuePairs

This is not ideal because it means this function is making a lot of assumptions about data existing out in the global scope that it doesn't have very much control over.

Ideally, this method would accept the `taskReport` as an argument, and then no globals would need to be accessed. 

Is this something that you can help me clean up?